### PR TITLE
Check metallb addon in user cluster for IPAM e2e tests

### LINKED ISSUE #10457: Extend IPAM e2e tests to check metallb allocations in user cluster

### PR DESCRIPTION
**Pull request by hbernardo**
**What does this PR do / Why do we need it**:
This PR extends the IPAM e2e tests to also check for the metallb addon allocation in the user cluster.

**Does this PR close any issues?**:
Fixes #10457

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -137,6 +136,8 @@ replace (
 	k8s.io/metrics => k8s.io/metrics v0.24.2
 )
 
+require go.universe.tf/metallb v0.13.3
```
Comment (xrstf) [inline go.mod]: I know I'm pedantic, but can you please put this requirement into the existing block of direct dependencies? :)) No idea why Go puts them into new `require` statements sometimes..

File: hack/ci/run-ipam-e2e-tests.sh [hunk 2]
```diff
@@ -39,7 +39,37 @@ protokol --kubeconfig "$KUBECONFIG" --flat --output "$ARTIFACTS/logs/kubermatic"
...
+kind: User
+metadata:
+  name: roxy-admin
+spec:
+  admin: true
+  email: roxy-admin@kubermatic.com
+  name: roxy-admin
+EOF
+retry 2 kubectl apply -f user.yaml
+
```
Comment (xrstf) [inline hack/ci/run-ipam-e2e-tests.sh]: Why do you create a roxy-admin user? The IPAM tests are headless, so there is no API/Dex. What am I missing? :D

Comment (hbernardo) [reply hack/ci/run-ipam-e2e-tests.sh]: No need for it, indeed. I will remove it

**Review by xrstf — APPROVED** [review]
/approve

**Comment by imharshita** [comment]
/hold
temporarily

**Pull request merged**